dotCMS/src/main/java/com/dotcms/filters/interceptor/saml/SamlWebInterceptor.java

```diff
@@ -0,0 +1,682 @@
+
+        final Map<String, Object> sessionAttributes = new HashMap<>();
+
+        if (null != currentSession && !currentSession.isNew()) {
+
+            Logger.debug(this, ()-> "Renewing the HTTP session");
+
+            attributesNames = currentSession.getAttributeNames();
+
+            while (attributesNames.hasMoreElements()) {
+
+                attributeName  = attributesNames.nextElement();
+                attributeValue = currentSession.getAttribute(attributeName);
+                Logger.debug(this, "Copying attribute '" + attributeName + "' to the new session.");
+                sessionAttributes.put(attributeName, attributeValue);
+            }
+
+            Logger.debug(this, ()->"Killing the current session");
+            currentSession.invalidate(); // kill the previous session
+
+            Logger.debug(this, ()->"Creating a new one");
+            renewSession = request.getSession(true);
+
+            for (final Map.Entry<String, Object> sessionEntry : sessionAttributes.entrySet()) {
+
+                Logger.debug(this, ()->"Adding attribute '" + sessionEntry.getKey() + "' to the new session.");
+                renewSession.setAttribute(sessionEntry.getKey(), sessionEntry.getValue());
+            }
+
+        }
+
+        return renewSession;
+    }
+
+    /**
+     * Perform the logic to get or create the user from the SAML and DotCMS If
+     * the SAML_ART_PARAM_KEY, will resolve the Assertion by calling a Resolver
+     * and will create/get/update the user on the dotcms data.
+     *
+     * @param request {@link HttpServletRequest}
+     * @param identityProviderConfiguration {@link IdentityProviderConfiguration}
+     * @return User
+     * @throws IOException
+     * @throws JSONException
+     * @throws DotDataException
+     */
+    public User getUser(final HttpServletRequest request,
+                        final IdentityProviderConfiguration identityProviderConfiguration) {
+
+        User systemUser = null;
+        User user       = null;
+
+        final HttpSession session = request.getSession(false);
+        if (session == null || request.isRequestedSessionIdValid()) {
+
+            Logger.error(this, "Could not retrieve user from session as the session doesn't exist!");
+            return null;
+        }
+
+        final String samlUserIdAttribute = identityProviderConfiguration.getId() + SAML_USER_ID;
+        if (null == session.getAttribute(samlUserIdAttribute)) {
+            return null;
+        }
+
+        final String samlUserId = (String) session.getAttribute(samlUserIdAttribute);
+        session.removeAttribute(identityProviderConfiguration.getId() + SAML_USER_ID);
+
+        try {
+            systemUser = this.userAPI.getSystemUser();
+            user       = this.userAPI.loadUserById(samlUserId, systemUser, false);
+        } catch (NoSuchUserException e) {
+
+            Logger.error(this, "No user matches ID '" + samlUserId + "'. Creating one...");
+            user = null;
```

Comment:
Codacy found an issue: [Assigning an Object to null is a code smell.  Consider refactoring.](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5707914)